decidim-core/app/packs/src/decidim/redesigned_input_character_counter.js

```diff
@@ -0,0 +1,288 @@
+// characters, this ratio is omitted and the announce threshold is always set to
+// 10.
+const SR_ANNOUNCE_THRESHOLD_RATIO = 0.1;
+// The number of characters left after which every keystroke will be announced.
+const SR_ANNOUNCE_EVERY_THRESHOLD = 10;
+const DEFAULT_MESSAGES = {
+  charactersAtLeast: {
+    one: `at least ${COUNT_KEY} character`,
+    other: `at least ${COUNT_KEY} characters`
+  },
+  charactersLeft: {
+    one: `${COUNT_KEY} character left`,
+    other: `${COUNT_KEY} characters left`
+  }
+};
+let MESSAGES = DEFAULT_MESSAGES;
+
+export default class InputCharacterCounter {
+  static configureMessages(messages) {
+    MESSAGES = $.extend(DEFAULT_MESSAGES, messages);
+  }
+
+  constructor(input) {
+    this.$input = input;
+    this.$target = $(this.$input.data("remaining-characters"));
+    this.minCharacters = parseInt(this.$input.attr("minlength"), 10);
+    this.maxCharacters = parseInt(this.$input.attr("maxlength"), 10);
+    this.describeByCounter = typeof this.$input.attr("aria-describedby") === "undefined";
+
+    // Define the closest length for the input "gaps" defined by the threshold.
+    if (this.maxCharacters > 10) {
+      if (this.maxCharacters > 100) {
+        this.announceThreshold = Math.floor(this.maxCharacters * SR_ANNOUNCE_THRESHOLD_RATIO / 100) * 100;
+      } else {
+        this.announceThreshold = 10;
+      }
+
+      // The number of characters left after which every keystroke will be announced.
+      this.announceEveryThreshold = SR_ANNOUNCE_EVERY_THRESHOLD;
+    } else {
+      this.announceThreshold = 1;
+      this.announceEveryThreshold = 1;
+    }
+
+    let targetId = this.$target.attr("id");
+    if (typeof targetId === "undefined") {
+      if (this.$input.attr("id") && this.$input.attr("id").length > 0) {
+        targetId = `${this.$input.attr("id")}_characters`;
+      } else {
+        targetId = `characters_${Math.random().toString(36).substr(2, 9)}`;
+      }
+    }
+
+    if (this.$target.length > 0) {
+      this.$target.attr("id", targetId)
+    } else {
+      const span = document.createElement("span")
+      span.id = targetId
+      span.className = "input-character-counter__text"
+
+      this.$target = $(span)
+
+      const div = document.createElement("div")
+      div.className = "input-character-counter__container"
+      div.appendChild(span)
+
+      // If input is a hidden for WYSIWYG editor add it at the end
+      if (this.$input.parent().is(".editor")) {
+        this.$input.parent().after(this.$target);
+      } else {
+        const wrapper = document.createElement("div")
+        wrapper.className = "input-character-counter"
+
+        // The form errors need to be in the same container with the field they
+        // belong to for Foundation Abide to show them automatically.
+        this.$input.next(".form-error").addBack().wrapAll(wrapper)
+        this.$input.after(div);
+      }
+    }
+
+    this.updateInputLength();
```

Comment:
I recently found one bug with the character counter which I am fixing at #9680. The fix is still pending for review at the moment.

Could you apply those changes to the redesigned version as well so that we won't re-introduce the bug here?

Reply:
Take a look if I apply the changes properly https://github.com/decidim/decidim/pull/9469/commits/16a0d1df4f440090ca512ca0612326b2257ea023

> The fix is still pending for review at the moment.

Do you want me to review it? The faster we merge it, the fewer bugs we'll introduce.

Reply:
Yes looks correct to, thanks!

With the review, go ahead if you want to, that would be appreciated. But we have a bit more general problem with the reviews being blocked right now that I hope we can resolve that once the holiday season ends and we get more maintainers working actively on the core. Our priority is now with RC2 with the backports, so those bug fix PRs will have to wait a bit.